src/Symfony/Component/OptionsResolver/Exception/NoConfigurationException.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver\Exception;
+
+use Symfony\Component\OptionsResolver\Debug\OptionResolverIntrospector;
+
+/**
+ * Thrown when trying to introspect an option definition property
+ * for which no value was configured inside the OptionResolver instance.
+ *
+ * @see OptionResolverIntrospector
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class NoConfigurationException extends \RuntimeException implements ExceptionInterface
```

Comment:
`UnconfiguredOptionException` sounds better to me and named consistently like the others?

Reply:
Same as for https://github.com/symfony/symfony/pull/24208#discussion_r141132091: Mentioning "option" right after "unconfigured" let us think the whole option is not set, whereas it's only part of the option definition which isn't. :/